### PR TITLE
disk_encryption macOS, fix for issue #4479

### PR DESCRIPTION
**Pull request by mkareta**
Fix for issue #4479 
We can not always determinate if disk is encrypted or not, so by using bool value (1/0) we losing potentially important information. In this PR expect reverting to old behaviour broken in recent PR, we also introduce encryption_status column that provide more context.

### CODE REVIEW

**Comment by theopolis** [comment]
If you only need a tri-state for the status I’d recommend {-1, 0, 1} where -1 indicates an error or undefined in this case. We use this in a few other places and I’d rather unixy.

**Comment by theopolis** [comment]
Then you don’t need the new column.

**Comment by mkareta** [comment]
There are few problems with -1: 
- it's breaks backward compatibility
- "-1" is not unified across project and provides less context than "undefined", because -1 may means data is unavailable, error happened or we can't determinate is disk encrypted or not
I think that general rule that we need to follow is: don't use numbers where they not represent numeric value. What do you think? 
CC: @fmanco @muffins @obelisk

File: osquery/tables/system/linux/disk_encryption.cpp [hunk 1]
```diff
@@ -23,6 +23,12 @@ extern "C" {
 
 #include "osquery/core/conversions.h"
 
+// FIXME: Add enum generation for tables and remove following code
+// Copy of values in disk_encryption.mm
+const std::string kEncryptionStatusEncrypted = "encrypted";
+const std::string kEncryptionStatusUndefined = "undefined";
+const std::string kEncryptionStatusNotEncrypted = "not encrypted";
```
Comment (akindyakov) [inline osquery/tables/system/linux/disk_encryption.cpp]: Looks like copy-paste  from `disk_encryption.mm`. Is here any way to use common code for it?

**Comment by muffins** [comment]
Yeah we had had a discussion around the different numerics in the code base and what they should mean. iirc we had arrived at leaving it to the table implementation to define what the numbers should be? @mkareta could we use `-1`, `0`, and `1` and just define what they mean in the spec? I don't have much preference either way, and I can't at the moment think of other places in the code base that also use `-1` for "error" or "undefined", so I agree in that case we might be breaking backwards compat, as I believe we've typically used `1` for error.

**Comment by obelisk** [comment]
I'm ok with drawing a line in sand and marking this API Change so we can be in a better place. Old values can still be emulated by using an updated query

File: osquery/tables/system/linux/disk_encryption.cpp (hunk 1 above)
Comment (mkareta) [reply osquery/tables/system/linux/disk_encryption.cpp]: It's copy-paste, this why there is FIXME there :)
It is very common use case so we need to solve it in general way, IMHO by providing enum generation as part of table generation

**Comment by mkareta** [comment]
I don't see a reason to break backward compatibility here and force users to update queries, I think we should be careful with such changes, and provide notice before removing columns. Anyway in this case support of backward compatibility costs is 0

**Pull request merged**